src/rules/unit-whitelist/index.js

```diff
@@ -28,14 +29,17 @@ export default function (whitelistInput) {
       const { value } = decl
 
       valueParser(value).walk(function (node) {
-        if (node.type === "function" && node.value.toLowerCase() === "url") { return false }
-        if (node.type !== "word" || !isStandardValue(node.value)) { return }
-
-        const parsedUnit = valueParser.unit(node.value)
+        // Remove interpolation characters
+        const value = blurInterpolation(node.value, "")
+        const valueLowerCase = value.toLowerCase()
 
-        if (!parsedUnit) { return }
+        if (node.type === "function" && valueLowerCase === "url") { return false }
+        if (node.type !== "word"
+          || !isStandardValue(value)
+          || !isFinite(parseInt(value))
+        ) { return }
 
-        const unit = parsedUnit.unit
+        const unit = getUnitFromValueNode(value)
```

Comment:
I'd suggest that you pass `node` as this argument, and include the rest of the repeated logic above in `getUnitFromValueNode()`. Does that make sense?